src/components/data-table/ha-data-table.ts

```diff
@@ -227,10 +227,13 @@ export class HaDataTable extends LitElement {
               const sorted = key === this._sortColumn;
               const classes = {
                 "mdc-data-table__header-cell--numeric": Boolean(
-                  column.type && column.type === "numeric"
+                  column?.type === "numeric"
```

Comment:
null-chaining is not cheap, it means that it rewrites it to `column && column.type === "numeric"`. So is this necessary? On line 245 we just assume `column` exists.